### PR TITLE
[Improvement][SPI]Migrate related classes loaded by SPI classes to public modules

### PR DESCRIPTION
**Pull request by CalvinKirs**

### CODE REVIEW

**Comment by gaojun2048** [comment]
SonarCloud Code Analysis failed

File: pom.xml [hunk 1]
```diff
@@ -563,21 +563,18 @@
                 <groupId>org.sonatype.aether</groupId>
                 <artifactId>aether-api</artifactId>
                 <version>1.13.1</version>
-                <scope>provided</scope>
```
Comment (EricJoy2048) [inline pom.xml:567]: Why change this "scope" ?

File: pom.xml [hunk 2]
```diff
@@ -563,21 +563,18 @@
...
             </dependency>
 
             <dependency>
                 <groupId>io.airlift.resolver</groupId>
                 <artifactId>resolver</artifactId>
                 <version>1.5</version>
-                <scope>provided</scope>
```
Comment (EricJoy2048) [inline pom.xml:574]: aether-api, resolver and asm only used in local development mode, used to load the plug-in through maven to load the plug-in pom.xml file configured in alert.properties to facilitate debugging during development. Not required when the server deployment is running.

File: dolphinscheduler-alert/src/main/java/org/apache/dolphinscheduler/alert/AlertServer.java [hunk 3]
```diff
@@ -20,8 +20,8 @@
 import static org.apache.dolphinscheduler.common.Constants.ALERT_RPC_PORT;
 
 import org.apache.dolphinscheduler.alert.plugin.AlertPluginManager;
-import org.apache.dolphinscheduler.alert.plugin.DolphinPluginLoader;
-import org.apache.dolphinscheduler.alert.plugin.DolphinPluginManagerConfig;
+import org.apache.dolphinscheduler.common.plugin.DolphinPluginLoader;
+import org.apache.dolphinscheduler.common.plugin.DolphinPluginManagerConfig;
```
Comment (chengshiwen) [inline dolphinscheduler-alert/src/main/java/org/apache/dolphinscheduler/alert/AlertServer.java:24]: These lines will resulting **import order** problems and **check style failure**

File: dolphinscheduler-alert/src/test/java/org/apache/dolphinscheduler/alert/AlertServerTest.java [hunk 4]
```diff
@@ -18,8 +18,8 @@
 package org.apache.dolphinscheduler.alert;
 
 import org.apache.dolphinscheduler.alert.plugin.AlertPluginManager;
-import org.apache.dolphinscheduler.alert.plugin.DolphinPluginLoader;
-import org.apache.dolphinscheduler.alert.plugin.DolphinPluginManagerConfig;
+import org.apache.dolphinscheduler.common.plugin.DolphinPluginLoader;
+import org.apache.dolphinscheduler.common.plugin.DolphinPluginManagerConfig;
```
Comment (chengshiwen) [inline dolphinscheduler-alert/src/test/java/org/apache/dolphinscheduler/alert/AlertServerTest.java:22]: import order problem

File: dolphinscheduler-alert/src/test/java/org/apache/dolphinscheduler/alert/plugin/AlertPluginManagerTest.java [hunk 5]
```diff
@@ -60,6 +62,6 @@ public void testLoadPlugins() throws Exception {
             throw new RuntimeException("load Alert Plugin Failed !", e);
         }
 
-        Assert.assertNotNull(alertPluginManager.getAlertChannelFactoryMap().get("email alert"));
+        Assert.assertNotNull(alertPluginManager.getAlertChannelFactoryMap().get("Email"));
```
Comment (chengshiwen) [inline dolphinscheduler-alert/src/test/java/org/apache/dolphinscheduler/alert/plugin/AlertPluginManagerTest.java:65]: Avoid the magic number `Email`, and it's better to use a **Constant**

File: dolphinscheduler-alert/src/test/java/org/apache/dolphinscheduler/alert/plugin/DolphinPluginLoaderTest.java [hunk 6]
```diff
@@ -55,6 +48,6 @@ public void testLoadPlugins() throws Exception {
             throw new RuntimeException("load Alert Plugin Failed !", e);
         }
 
-        Assert.assertNotNull(alertPluginManager.getAlertChannelFactoryMap().get("email alert"));
+        Assert.assertNotNull(alertPluginManager.getAlertChannelFactoryMap().get("Email"));
```
Comment (chengshiwen) [inline dolphinscheduler-alert/src/test/java/org/apache/dolphinscheduler/alert/plugin/DolphinPluginLoaderTest.java:51]: Avoid the magic number `Email`

**Review by chengshiwen — CHANGES_REQUESTED** [review]

File: dolphinscheduler-alert/src/test/java/org/apache/dolphinscheduler/alert/plugin/DolphinPluginLoaderTest.java (hunk 6 above)
Comment (CalvinKirs) [reply dolphinscheduler-alert/src/test/java/org/apache/dolphinscheduler/alert/plugin/DolphinPluginLoaderTest.java:51]: The alert module does not depend on specific plug-in modules, so defining constants here may not be a good solution.

**Review by gaojun2048 — APPROVED** [review]
+1

**Review by lgcareer — APPROVED** [review]
+1

**Pull request merged**